vignettes/wip/dm-function-naming-logic.Rmd

```diff
@@ -0,0 +1,43 @@
+---
+title: "Function naming logic"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+This vignette describes the logic behind the names of the {dm}-functions.
+
+In general the functions are implemented in such a way, that no function can handle both `dm` classes (`dm` and `zoomed_dm`) as an argument.
+A `zoomed_dm` behaves in most situations like a table and there are mostly no explicit functions available for this class, but rather methods (see vignette ["Zooming and manipulating tables"]().
+<!-- FIXME: Vignette needs to be reviewed and integrated (https://github.com/krlmlr/dm/issues/185) -->
+
+## Prefixes
+
+1. `dm_`: There are two cases for this prefix:
+    - The function takes a `dm` object as a first argument
+    - The function constructs a `dm` object from other objects
```

Comment:
Which functions do this, short of `dm()`?

Reply:
`dm_from_src()`, `dm_nycflights13()` and `dm_learn_from_db()` (didn't we consider changing its name to `learn_dm_from_db()`?)